src/ServiceManager.php

```diff
@@ -67,16 +68,13 @@ class ServiceManager implements ServiceLocatorInterface
 
     /**
      * Whether or not changes may be made to this instance.
-     *
-     * @var bool
      */
-    protected $allowOverride = false;
+    protected bool $allowOverride = false;
```

Comment:
@boesing it seems make error in `pslam --stepherd`

```
Error: Type false for $this->allowOverride is always falsy (see https://psalm.dev/122)
```

Do you have idea to handle it?

Reply:
Not really. 

Maybe the reason why this is an error is, that there is no way of changing `allowOverride` via codepath as it only is overridden in plugin managers.
So psalm thinks "this will never change to an other value than `false`.

But thats a weird assumption as it is `protected`. Maybe thats a bug?

Reply:
probably, as add typed parameter in `setAllowOverride()` and `getAllowOverride()` seems stil got the error:

![Screen Shot 2021-09-16 at 21 33 00](https://user-images.githubusercontent.com/459648/133631224-28906178-5456-4e9f-962a-7c00248ab674.png)

Reply:
any way we can ignore the error?

Reply:
Ah, the issue is that `allowOverride` is already verified in the method earlier:

```
if ($this->allowOverride || ! $this->configured) {
    return;
}
```

So thats why you can just remove these checks at all.

Reply:
Fixed 🎉 https://github.com/laminas/laminas-servicemanager/pull/104/commits/6d2438235fcec2792e069bc9eb236647cebca5ad thank you